### DISCUSSION TITLE
Question About StreamOpen

### DISCUSSION THREAD

**Discussion by zdh3303** [microsoft/msquic#2121] [Q&A] [answered]
hi, when I use StreamOpen with flag `QUIC_STREAM_OPEN_FLAG_0_RTT` on the server side, however on the other side, I receive the Event `QUIC_CONNECTION_EVENT_PEER_STREAM_STARTED` , however `QUIC_STREAM_OPEN_FLAGS type = Event->PEER_STREAM_STARTED.Flags;` this value is zero (`QUIC_STREAM_OPEN_FLAG_NONE`), not same as the server side's.

**Comment by nibanks** [comment, marked as answer]
That flag, `QUIC_STREAM_OPEN_FLAG_0_RTT`, is merely a hint to the API that says "This stream data is allowed to be sent in 0-RTT". It doesn't mean it will always use 0-RTT. A number of conditions have to happen first, primarily that the connection actually supports and uses TLS resumption and 0-RTT. My guess is that is not happening.

To use 0-RTT, you must use OpenSSL TLS (Schannel doesn't support it yet). Also, you must initially connect to the server and get a resumption ticket (received via the `QUIC_CONNECTION_EVENT_RESUMPTION_TICKET_RECEIVED` event). Then, on your next connection to the server, supply that token via `SetParam` with the `QUIC_PARAM_CONN_RESUMPTION_TICKET` parameter.

We don't have great docs or a sample for this yet. Might be a good idea to add both... @anrossi?

**Reply by zdh3303** [reply]
thanks, I have another question , I want to create two stream(each stream has its own function) , how can I let the other side seperate the streams, and do some different works acording to the different stream

**Reply by nibanks** [reply]
> how can I let the other side seperate the streams

They will get two separate peer stream started events. They can then do whatever they want with them.
